### PR TITLE
python music scrapers - art

### PR DESCRIPTION
**Pull request by ronie**
this is my clumsy attempt to let the python music scrapers set some additional artwork.

(needs https://github.com/xbmc/xbmc/pull/11754)

@notspiff

### CODE REVIEW

File: xbmc/addons/Scraper.cpp [hunk 1]
```diff
@@ -751,6 +751,13 @@ void DetailsFromFileItem<CAlbum>(const CFileItem& item, CAlbum& album)
 
   int nThumbs = item.GetProperty("album.thumbs").asInteger();
   ParseThumbs(album.thumbURL, item, nThumbs, "album.thumb");
+
+  std::map<std::string, std::string> artwork;
+  std::string cdart = FromString(item, "album.cdart.url");
+  if (!cdart.empty())
+    artwork.insert(std::pair<std::string, std::string>("cdart", cdart));
```
Comment (notspiff) [inline xbmc/addons/Scraper.cpp]: std::make_pair("cdart", cdart)

**Comment by notspiff** [comment]
looks fine, better with make_pair

File: xbmc/addons/Scraper.cpp [hunk 2]
```diff
@@ -785,6 +792,16 @@ void DetailsFromFileItem<CArtist>(const CFileItem& item, CArtist& artist)
...
   artist.fanart.Unpack();
+
+  std::map<std::string, std::string> artwork;
+  std::string logo = FromString(item, "artist.clearlogo.url");
+  if (!logo.empty())
+    artwork.insert(std::pair<std::string, std::string>("clearlogo", logo));
+  std::string banner = FromString(item, "artist.banner.url");
+  if (!banner.empty())
+    artwork.insert(std::pair<std::string, std::string>("banner", banner));
+  if (!artwork.empty())
```
Comment (xhaggi) [inline xbmc/addons/Scraper.cpp]: Minor, but you could re-use the same var for every value if you only want to transfer it into the map.

```
std::string value = FromString(...
if (!value.empty())
  // transfer value into map
value = FromString(...
if (!value.empty())
  // transfer value into map

```

**Comment by Razzeee** [comment]
@ronie 
do you want to adapt this before we merge?

**Comment by ronie** [comment]
sure, i'll look into it asap.. hopefully next week

**Comment by jjd-uk** [comment]
Does this still need changes?

@DaveTBlake for review as well

**Comment by ronie** [comment]
> Does this still need changes?

updated now with the suggestions made above

**Comment by DaveTBlake** [comment]
Any reason for adding "cdart" for albums, and "banner" and "clearlogo" for artists?

**Comment by zag2me** [comment]
Back Album cover and Spine album cover would also be nice. Both are supported by TheAudioDB and have already been integrated into at least one skin by hardcoding the filenames.

See: http://forum.kodi.tv/showthread.php?tid=313352&pid=2586744#pid2586744

**Comment by DaveTBlake** [comment]
Is there an easy way to make the suppported artwork pairs configurable rather than hard coded?

**Comment by zag2me** [comment]
This discussion has come up before and its perfectly legitimate and sensible request. But I have always believed that artwork types should be hard coded to provide consistency through different skins and Add-ons.

**Comment by phil65** [comment]
They shouldnt be hardcoded, but we should have some conventions which ones are used commonly.

**Comment by DaveTBlake** [comment]
Fair enough @zag2me I can see your point, but at the moment it all seems very arbitrary and inconsistent to me.  @phil65 a convention would be welcome.

I guess "cdart" is general for albums, or is it an image of the disk? In comparison "cover" (front), "back" and "spine" are clearly specific. Then there are also multiple images from any  liner booklet art, do we go that far?

For artist how are "banner" and "clearlogo" used, or compare to fanart?

**Comment by phil65** [comment]
banner and clearlogo are the same as for movies. cdart is an image of the cd/dvd itself.

**Comment by MartijnKaijser** [comment]
See https://fanart.tv/  how they look.
Videos database support the standard poster/fanart as hardcoded and with JSON-RPC you can additionally add whatever image names you want by simply defining them.

**Comment by DaveTBlake** [comment]
Ah much clearer, thanks. I tend to listen to music with TV off so oblivious to all this glorious art!

**Comment by rmrector** [comment]
`ParseThumbs` just above these can already pull in all freely named artwork for Python scrapers, by setting "aspect" to "clearlogo" or whatever. "aspect" is a bit of a misnomer but in the video library it is directly translated to "art type" when it is added (see #12612 for the missing artwork there), and `ParseThumbs` is shared between video and music, so extended artwork in the music library isn't being held back here.

**Comment by ronie** [comment]
superseded by https://github.com/xbmc/xbmc/pull/13848

**Pull request closed without merging**